### PR TITLE
Fix omitting of optional record fields

### PR DESCRIPTION
**Pull request by davidyuk**

### CODE REVIEW

**Review by dincho — CHANGES_REQUESTED** [review]
Please move the implementation to the InternalMapper only. The internal structures should not be affected by this changes.

Also, please add release notes.

File: tests/integration/encoder.sh [hunk 1]
```diff
@@ -55,6 +55,7 @@ test_encoder 'test_template_maze(Any({origin = {x = 1, y = 2}, a = 3, b = 4}, Ye
 test_encoder 'test_record({x = 0, y = 0})'
 test_encoder 'test_singleton_record({x = 0})'
 test_encoder 'test_nested_record({origin = {x = 1, y = 2}, a = 3, b = 4})'
+test_encoder 'test_option_record({x = 42, y = None})'
```
Comment (dincho) [inline tests/integration/encoder.sh:58]: Does the CLI compiler works without providing the optional ?

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 - Fix encoding and decoding of singular tuples and records (optimization)
 - Fix support of templated records
+- Fix omitting of optional record fields
```
Comment (davidyuk) [inline CHANGELOG.md:12]: Not sure about this, omitting of optional fields was broken after the previous release (at the same time as `Set` support)

Comment (davidyuk) [reply CHANGELOG.md:12]: Yep, I found it while running sdk tests with the calldata from master branch

**Pull request merged**